core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerService.js

```diff
@@ -189,6 +202,22 @@ oppia.factory('SimpleEditorManagerService', [
             SimpleEditorShimService.saveAnswerGroups(
               currentStateName, newAnswerGroups);
           }
+          if (index === questionCount - 1 && doesLastQuestionHaveAnswerGroups) {
```

Comment:
I'm not sure I understand why this was changed. Please could you clarify? It seems to me like you're creating some kind of self-loop here rather than a link to the next question.

Reply:
We're facing an error while changing type the for the last question which has an answer group. This particular case was missed, at this point, nextState doesn't exist and thus we ended getting an error from Outcome Object Factory.